### PR TITLE
Updating service.json to add few properties to VMNicDetails and VMNicInputDetails

### PR DESCRIPTION
**Pull request by punit-bhatt**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -16717,6 +16717,21 @@
...
+        "recoveryPublicIpAddressId": {
+          "description": "The id of the public IP address resource associated with the NIC.",
+          "type": "string"
+        },
+        "recoveryNetworkSecurityGroupId": {
+          "description": "The id of the NSG associated with the NIC.",
+          "type": "string"
+        },
+        "recoveryLBBackendAddressPoolIds": {
+          "description": "The target backend address pools for the nic.",
```
Comment (yungezz) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: `NIC` insteaf of `nic` for consistence.

**Comment by yungezz** [comment]
@punit1396 pls fix ModelValidator error and Javascript SDK error in CI  https://dev.azure.com/azure-sdk/public/_build/results?buildId=53142

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 1 above)
Comment (punit1396) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: Updated.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 2]
```diff
@@ -16717,6 +16717,21 @@
           "description": "IP allocation type for recovery VM.",
           "type": "string"
         },
+        "recoveryPublicIpAddressId": {
```
Comment (majastrz) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: >recoveryPublicIpAddressId [](start = 9, length = 25)

You are modifying a stable API version by adding extra properties, which is considered a breaking change. These types of changes should be introduced in a new API version.

**Review by majastrz — CHANGES_REQUESTED** [review]
You are modifying a stable API version by adding extra properties, which is considered a breaking change. These types of changes should be introduced in a new API version.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 2 above)
Comment (punit1396) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: As per [api-guidelines/definition-of-breaking-change](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change), these changes do not qualify as a breaking change. Changes aren't made to the existing parameters and neither are these mandatory. Hence, there's no issue related to the backward compatibility.

Comment (sergey-shandar) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: IMHO, it only applies for `required` properties.

Comment (majastrz) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: Yeah, I missed the `required` aspect and assumed the worst. Sorry for the confusion.

---
In reply to: [296965293](https://github.com/Azure/azure-rest-api-specs/pull/6380#discussion_r296965293) [](ancestors = 296965293)

**Review by majastrz — APPROVED** [review]

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**